tools/gn

```diff
@@ -196,10 +196,14 @@ def to_gn_args(args):
 
     gn_args['dart_custom_version_for_pub'] = 'flutter'
 
-    # Make sure host_cpu matches the bit width of target_cpu.
+    # Make sure host_cpu matches the bit width of target_cpu. The only exception
+    # is that 32-bit Mac binaries are deprecated, so we switch to simarm_x64.
     target_is_32_bit = gn_args['target_cpu'] == 'arm' or gn_args['target_cpu'] == 'x86'
     if target_is_32_bit:
-      gn_args["host_cpu"] = "x86"
+      if sys.platform == 'darwin' and gn_args['target_cpu'] == 'arm':
```

Comment:
I expected `i386` variants of `gen_snapshot` to not be required for Linux either. Is this not the case?

Reply:
I suppose so, but I haven't tested that case, and I was trying to limit the scope of this change. Is there a reason to switch linux to simarm_x64?

Reply:
> Is there a reason to switch linux to simarm_x64?

We maintain a legacy toolchain for this purpose on Linux. This is preventing compiler updates. We were hoping to upgrade the same expecting this set of updates.

Reply:
Ok. I'll work on testing this on Linux using a 32 bit Android.

Reply:
Thanks. I appreciate it. This will really help Flutter simplify its toolchains and fix numbers hacks & workarounds.

Reply:
I've tested some example apps from my Linux machine to a Moto G4 and it's working.